examples/java/src/main/java/org/apache/beam/examples/cookbook/IcebergRestCatalogStreamingWriteExample.java

```diff
@@ -0,0 +1,203 @@
+    final String pubsubTopic = options.getTopic();
+    final String catalogUri = options.getCatalogUri();
+    final String warehouseLocation = options.getWarehouse();
+    final String projectName = options.getProject();
+    final String catalogName = options.getCatalogName();
+    final int triggeringFrequencySeconds = 30;
+
+    // Note: The token expires in 1 hour, and users may need to re-run the pipeline.
+    // Upcoming changes in Iceberg and the BigLake Metastore with the Iceberg REST Catalog
+    // will support token refreshing and credential vending.
+    Map<String, String> catalogProps =
+        ImmutableMap.<String, String>builder()
+            .put("type", "rest")
+            .put("uri", catalogUri)
+            .put("warehouse", warehouseLocation)
+            .put("header.x-goog-user-project", projectName)
+            .put("oauth2-server-uri", "https://oauth2.googleapis.com/token")
+            .put(
+                "token",
+                GoogleCredentials.getApplicationDefault().refreshAccessToken().getTokenValue())
+            .put("rest-metrics-reporting-enabled", "false")
+            .build();
+
+    Map<String, Object> icebergWriteConfig =
+        ImmutableMap.<String, Object>builder()
+            .put("table", tableIdentifier)
+            .put("catalog_properties", catalogProps)
+            .put("catalog_name", catalogName)
+            .put("triggering_frequency_seconds", triggeringFrequencySeconds)
+            .build();
+
+    Pipeline p = Pipeline.create(options);
+
+    PCollection<Row> aggregatedRows =
+        p.apply("ReadFromPubSub", PubsubIO.readStrings().fromTopic(pubsubTopic))
+            .apply("ConvertJsonToRow", JsonToRow.withSchema(TAXIRIDES_SCHEMA))
+            .apply(
+                "FilterNullFields",
+                Filter.by(
+                    (Row row) ->
+                        row.getInt64("passenger_count") != null
+                            && row.getDateTime("timestamp") != null))
+            .apply(
+                "FilterDropoffRides",
+                Filter.by((Row row) -> "dropoff".equals(row.getString("ride_status"))))
+            .apply(
+                "ApplyFixedWindow", Window.<Row>into(FixedWindows.of(Duration.standardSeconds(10))))
+            .apply(
+                "ExtractMinuteAndPassengerCount",
+                MapElements.into(
+                        TypeDescriptors.kvs(TypeDescriptors.strings(), TypeDescriptors.longs()))
+                    .via(
+                        (Row row) -> {
+                          Long passengerCount = row.getInt64("passenger_count");
+                          DateTime timestamp = (DateTime) row.getDateTime("timestamp");
+                          if (timestamp == null) {
+                            throw new IllegalStateException(
+                                "Timestamp should not be null at this stage.");
+                          }
```

Comment:
I think we don't need this null check.

If it's giving you a null warning, just wrap `row.getDateTime()` in a  `Preconditions.checkStateNotNull()`